src/freertos_drivers/net_cc32xx/CC32xxSocket.cxx

```diff
@@ -679,8 +680,14 @@ int CC32xxSocket::close(File *file)
     if (--references_ == 0)
     {
         mutex.unlock();
-        /* request that the socket be closed */
-        CC32xxWiFi::instance()->select_wakeup(s->sd);
+        /* request that the socket be removed from managment */
+        int16_t sd = s->sd;
+        CC32xxWiFi::instance()->fd_remove(sd);
+        portENTER_CRITICAL();
+        remove_instance_from_sd(sd);
```

Comment:
There are five different delete functions by now:

- del_socket
- fd_remove
- remove_instance_from_sd
- Socket::close()
- ~CC32xxSocket()

which are calling each other in hard to understand patterns that are not indicated by their respective names and namespaces. This is usually a sign that the code has organically evolved and it would be time to refactor it, specifically to find the appropriate concepts, encapsulation and naming scheme.

This is not something you should do in this PR, but file a bug and get to it whenever you have time.

If you'd like we could go over the use-cases in a chat to find the appropriate encapsulation concepts that can make the code more readable.